add: `View Branch on GitHub` Context Menu Item

DylanDevelops opened the pull request:
<!--
What GitHub Desktop issue does this PR address (for example, #1234)?
If you have not created an issue for your PR, please search the issue tracker to see if there is an existing issue that aligns with your PR, or open a new issue for discussion.
-->

Closes #21071

## Description
<!--
A summary of the changes made along with any other information that would be helpful to a reviewer such as potential tradeoffs or alternative approaches you considered.
-->
I added the ability to open the branch on GitHub if it exists on the remote. I accomplished this by crafting a URL with parameters retrieved from the repository. This was the solution I came up with to craft this URL; however, if you know of a better way, let me know!

Additionally, I conditionally render the context menu based on whether the remote name variable exists. I'm not sure if this is the best way to do this, either, but it seems to work. Please let me know if there is an easier variable I can check.

### Screenshots

<!--
If this PR touches the UI layer of the app, please include screenshots or animated gifs to show the changes.
-->

Main Branch:
<img width="333" height="177" alt="Screenshot 2025-11-13 at 2 07 39 PM" src="https://github.com/user-attachments/assets/7f616dc0-9b79-4991-8ebb-28dad24ea908" />

Branch w/ Pull Request:
<img width="384" height="199" alt="Screenshot 2025-11-13 at 2 07 51 PM" src="https://github.com/user-attachments/assets/d65247b7-c85f-4f8f-b673-f61b2991ba96" />

Local Branch:
<img width="531" height="150" alt="Screenshot 2025-11-13 at 2 08 04 PM" src="https://github.com/user-attachments/assets/b327f8bb-45c6-4b48-a3c1-9224da3fa5f0" />

## Release notes

<!--
You can leave this blank if you're not sure.
If you don't believe this PR needs to be mentioned in the release notes, write "Notes: no-notes".
-->

Notes: [add] "View Branch on GitHub" option on branch list context menu

DylanDevelops committed:
Add 'View Branch on GitHub' to branch context menu

Introduces a new context menu item allowing users to view the current branch on GitHub if an upstream remote is set. Implements the handler in BranchDropdown and updates the context menu item generation logic.

DylanDevelops committed:
Fix typo in branch dropdown handler name

Renamed onViewBranchOnGithub to onViewBranchOnGitHub for consistency and to fix a typo in the branch dropdown component.

DylanDevelops committed:
Refactor branch dropdown GitHub view logic

Simplified variable usage in onViewBranchOnGitHub by directly accessing props and removing redundant assignments.

niik commented on app/src/ui/toolbar/branch-dropdown.tsx:
```diff
@@ -309,6 +309,9 @@ export class BranchDropdown extends React.Component<IBranchDropdownProps> {
       onRenameBranch: this.onRenameBranch,
+      onViewBranchOnGitHub: tip.branch.upstreamRemoteName
+        ? this.onViewBranchOnGitHub
+        : undefined,
```
This needs to check whether the repository is a GitHub repository (as opposed to a self-hosted, local, or a repository hosted on another Git provider).


```suggestion
      onViewBranchOnGitHub:
        isRepositoryWithGitHubRepository(this.props.repository) &&
        tip.branch.upstreamRemoteName
          ? this.onViewBranchOnGitHub
          : undefined,
```

niik commented on app/src/ui/toolbar/branch-dropdown.tsx:
```diff
@@ -338,6 +341,26 @@ export class BranchDropdown extends React.Component<IBranchDropdownProps> {
+      return
+    }
+
+    const branchName = tip.branch.upstreamWithoutRemote ?? tip.branch.name
```
Since we won't show the context menu item unless `upstreamWithourRemote` is set we should bail here if it isn't as well.

```suggestion
    if (!tip.branch.upstreamWithoutRemote) {
      return
    }

    const url = `${gitHubRepository.htmlURL}/tree/${encodeURIComponent(
      tip.branch.upstreamWithoutRemote
    )}`
```

niik reviewed (changes requested):
Thanks @DylanDevelops! Left two suggestions but happy to bring this in once that's resolved

DylanDevelops committed:
Update app/src/ui/toolbar/branch-dropdown.tsx

Co-authored-by: Markus Olsson <j.markus.olsson@gmail.com>

DylanDevelops committed:
Update app/src/ui/toolbar/branch-dropdown.tsx

Co-authored-by: Markus Olsson <j.markus.olsson@gmail.com>

DylanDevelops committed:
Fix duplicate URL assignment in BranchDropdown

Removed redundant assignment of the 'url' variable in the BranchDropdown component to ensure only the correct URL is used when opening a branch in the browser.

DylanDevelops commented:
@niik I implemented your suggested changes and fixed up the small formatting issues. When I tested, everything seemed to work great! Let me know if this looks good or if there is anything else I should do.

niik reviewed (approved):
Love it, thanks a bunch @DylanDevelops

DylanDevelops commented:
Thank you, @niik!
